cinema/tests/test_movie_api.py

```diff
@@ -157,3 +156,160 @@ def test_image_url_is_shown_on_movie_session_detail(self):
+        movie1 = sample_movie()
+        movie2 = sample_movie()
+        actor1 = sample_actor()
+        actor2 = sample_actor()
+
+        movie1.actors.add(actor1)
+        movie2.actors.add(actor2)
+
+        serializer1 = MovieListSerializer(movie1)
+        serializer2 = MovieListSerializer(movie2)
+
+        res = self.client.get(MOVIE_URL, {"actors": f"{actor1.id}"})
+
+        self.assertIn(serializer1.data, res.data)
+        self.assertNotIn(serializer2.data, res.data)
+
+    def test_retrieve_movie_detail(self):
+        movie = sample_movie()
+        url = detail_url(movie.id)
+        res = self.client.get(url)
+
+        serializer = MovieDetailSerializer(movie)
+
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEqual(res.data, serializer.data)
+
+    def test_create_movie_forbidden(self):
+        movie1 = {
+            "title": "test1",
+            "description": "Test1",
+            "duration": 90,
+        }
+        res = self.client.post(MOVIE_URL, movie1)
+        self.assertEqual(res.status_code, status.HTTP_403_FORBIDDEN)
+
+
+class AdminMovieApiTest(TestCase):
+    def setUp(self) -> None:
+        self.client = APIClient()
+        self.user = get_user_model().objects.create_user(
+            "admin@myproject.com",
+            "password",
+            is_staff=True
+        )
+        self.client.force_authenticate(self.user)
+
+    def test_create_movie(self):
+        genre = sample_genre()
+        actor = sample_actor()
+        new_movie = {
+            "title": "test1",
+            "description": "Test1",
+            "duration": 90,
+            "actors": [genre.id],
+            "genres": [actor.id]
+        }
+
+        res = self.client.post(MOVIE_URL, new_movie)
+        movie = Movie.objects.get(id=res.data["id"])
+        actors = movie.actors.all()
+        genres = movie.genres.all()
+
+        self.assertEqual(res.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(new_movie["title"], movie.title)
+        self.assertEqual(new_movie["description"], movie.description)
+        self.assertEqual(new_movie["duration"], movie.duration)
+        self.assertIn(actor, actors)
+        self.assertIn(genre, genres)
+
+    def test_delete_movie_not_allowed(self):
+        movie = sample_movie()
+
+        url = detail_url(movie.id)
+
+        res = self.client.delete(url)
+
+        self.assertEqual(res.status_code, status.HTTP_405_METHOD_NOT_ALLOWED)
+
+    def test_put_movie_not_allowed(self):
+        movie = sample_movie()
+
+        url = detail_url(movie.id)
+
+        res = self.client.put(url)
+
+        self.assertEqual(res.status_code, status.HTTP_405_METHOD_NOT_ALLOWED)
+
+    def test_patch_movie_not_allowed(self):
+        movie = sample_movie()
+
+        url = detail_url(movie.id)
+
+        res = self.client.patch(url)
+
+        self.assertEqual(res.status_code, status.HTTP_405_METHOD_NOT_ALLOWED)
+
+
+
+
```

Comment:
Lots of whitespaces inside the functions and in the end of the file. Use whitespaces to separate logical pars of the code and remove extra ones